### PR TITLE
merge 11.18.1-discord-2 (open source branch) into 11.18.1-discord-1 (fork used in Discord app)

### PR DESCRIPTION
**Pull request by donaldchen**
Discord is currently using this `11.18.1-discord-1` branch. On `11.18.1-discord-2`, I rebased from `master` a few times and began addressing comments / feedback in https://github.com/react-native-webview/react-native-webview/pull/2469 . In order to address remaining feedback, I'd like to

- (in this PR) Bring the changes from that PR to the `11.18.1-discord-1` branch.
- (in a future PR) Merge master with the new arch changes into `11.18.1-discord-1`.
- (in a future PR) Take that merge commit with the new arch changes and resolved conflicts on `11.18.1-discord-1` and cherry-pick it to `11.18.1-discord-2`.

Prior to this PR, `11.18.1-discord-1` had the following changes that `11.18.1-discord-2` did not:
- https://github.com/discord/react-native-webview/pull/24
- https://github.com/discord/react-native-webview/pull/23
- https://github.com/discord/react-native-webview/commit/dbf94193ad64d7d780842a25a24845ea7b4d17a2
- https://github.com/discord/react-native-webview/commit/237796aced1f986a5f8c33d5be295c5fbf74e728 .

I'm aiming to keep those changes on `11.18.1-discord-1` for now, and we can create PRs for those to the upstream repo in the future.

### CODE REVIEW

File: apple/RNCScriptMessageManager.m [hunk 1]
```diff
@@ -34,19 +34,19 @@ @interface RNCScriptMessageManager() {
...
 + (id) sharedManager {
-    static RNCScriptMessageManager *_sharedManager = nil;
-    @synchronized(self) {
-      if(_sharedManager== nil) {
-        _sharedManager = [[super alloc] init];
-      }
-      return _sharedManager;
+  static RNCScriptMessageManager *_sharedManager = nil;
+  @synchronized(self) {
+    if(_sharedManager== nil) {
```
Comment (nealmanaktola) [inline apple/RNCScriptMessageManager.m:39]: nit: `_sharedManager == nil`

File: src/WebViewProxy.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { EmitterSubscription, NativeEventEmitter } from "react-native";
+import { IWebViewProxy } from './WebViewProxyTypes';
+
+const scriptMessageEmitter = new NativeEventEmitter(undefined);
```
Comment (nealmanaktola) [inline src/WebViewProxy.ts:4]: does this work on `iOS`  - we used to pass the NativeModule before 

https://github.com/discord/react-native-webview/blob/11.18.1-discord-1/src/addOnMessageListenerWithWebViewKey.ts#L5

Comment (nealmanaktola) [reply src/WebViewProxy.ts:4]: oh nvm just saw `.ios.ts`

Comment (nealmanaktola) [reply src/WebViewProxy.ts:4]: ah - it seems like we have .ios.ts/.android.ts + the default

Comment (donaldchen) [reply src/WebViewProxy.ts:4]: Yup! That's right. I did test message passing in both directions between the RN app and the iframe in the WebView on iOS and Android, so I can confirm that this still works on iOS

File: apple/RNCScriptMessageManager.m (hunk 1 above)
Comment (donaldchen) [reply apple/RNCScriptMessageManager.m:39]: good catch! I'll update this

Comment (donaldchen) [reply apple/RNCScriptMessageManager.m]: https://github.com/discord/react-native-webview/pull/25/commits/d7a448927a365fc799c790a329fddb90034776e1

File: src/WebViewProxy.ts (hunk 2 above)
Comment (nealmanaktola) [reply src/WebViewProxy.ts:4]: 👍

**Pull request merged**